### PR TITLE
New feature: Related Posts

### PR DESCRIPTION
**Pull request by byanko55**
**New Features**
* Display related posts

**Minor Updates**
* Trivial changes in search bar

### CODE REVIEW

**Review by mgarrixx — CHANGES_REQUESTED** [review]
Bruh watch this

image

**Comment by mgarrixx** [comment]
And what if the category attribute is not given?

File: assets/js/main.js [hunk 1]
```diff
@@ -534,11 +539,91 @@ document.addEventListener('DOMContentLoaded', function(){
...
+            date = date.toLocaleString('en-US', {day: 'numeric', month:'long', year:'numeric'});
+
+            if (post.thumbnail === ''){
+                post.thumbnail = "/assets/img/thumbnail/empty.jpg";
+            }
+
+            $('#related-posts').append(
+                '<li class="related-item"><a href="' + post.url +
+                    '"><img src="' + post.thumbnail + 
+                    '"/><p class="category">' + post.category +
```
Comment (mgarrixx) [inline assets/js/main.js:606]: Please assure that rendering is not failed even when the `post.category` is blank.

Comment (mgarrixx) [reply assets/js/main.js:606]: I suspect, it could arise an error earlier in `line 562`

**Review by mgarrixx — CHANGES_REQUESTED** [review]

File: assets/js/main.js [hunk 2]
```diff
@@ -534,11 +539,91 @@ document.addEventListener('DOMContentLoaded', function(){
...
+                return b.score - a.score;
+            }
+        });
+
+        if (relatedPosts.length == 0){
+            $('#related-box').hide();
+            return;
+        }
+
+        for (var i = 0; i < Math.max(relatedPosts.length, 6); i++){
```
[the remark this answers is not in the document]
Comment (byanko55) [reply assets/js/main.js]: Trutly that is not the right process!

**Comment by byanko55** [comment]
I'd found it was far to merge; The additional updates I did:

* Handle the post on which user does not specify the posting date. Also, in that case, I marked '`-`' on date field (in both `pagination.html` and `post.html`).
* Fix the width of `#related-post` as 100%

**Review by cvlian — COMMENTED** [review]
You didn't put some codes adjusting element size in mobile view. right?

**Comment by byanko55** [comment]
@cvlian 
Honestly I have bunch of thing to do about style in mobile layout. So, let me handle them later all at once.

**Review by cvlian — APPROVED** [review]
> @cvlian Honestly I have bunch of thing to do about style in mobile layout. So, let me handle them later all at once.

Whatever, I'll leave it up to you.

**Review by mgarrixx — APPROVED** [review]
No warnings for now. I had confirmed.

**Pull request merged**